environment/filesystem.go

```diff
@@ -45,6 +48,81 @@ func (env *Environment) FileWrite(ctx context.Context, explanation, targetFile,
 	return nil
 }
 
+func (env *Environment) FileEdit(ctx context.Context, explanation, targetFile, search, replace, matchID string) error {
+	contents, err := env.container().File(targetFile).Contents(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Find all matches of the search text
+	matches := []int{}
+	searchIndex := 0
+	for {
+		index := strings.Index(contents[searchIndex:], search)
+		if index == -1 {
+			break
+		}
+		actualIndex := searchIndex + index
+		matches = append(matches, actualIndex)
+		searchIndex = actualIndex + 1
+	}
+
+	if len(matches) == 0 {
+		return fmt.Errorf("search text not found in file %s", targetFile)
+	}
+
+	// If there are multiple matches and no matchID is provided, return an error with all matches
+	if len(matches) > 1 && matchID == "" {
+		var matchDescriptions []string
+		for i, matchIndex := range matches {
+			// Generate a unique ID for each match
+			id := generateMatchID(targetFile, search, replace, i)
+
+			// Get context around the match (3 lines before and after)
+			context := getMatchContext(contents, matchIndex)
+
+			matchDescriptions = append(matchDescriptions, fmt.Sprintf("Match %d (ID: %s):\n%s", i+1, id, context))
+		}
+
+		return fmt.Errorf("multiple matches found for search text in %s. Please specify which_match parameter with one of the following IDs:\n\n%s",
+			targetFile, strings.Join(matchDescriptions, "\n\n"))
+	}
+
+	// Determine which match to replace
+	var targetMatchIndex int
+	if len(matches) == 1 {
+		targetMatchIndex = matches[0]
+	} else {
+		// Find the match with the specified ID
+		found := false
+		for i, matchIndex := range matches {
+			id := generateMatchID(targetFile, search, replace, i)
```

Comment:
how's the model supposed to decide which match ID to use in the case where there's multiple?